### DISCUSSION TITLE
Milvus Cluster Auto Scaling Support on GCP

### DISCUSSION THREAD

**Discussion by divyajaincs** [milvus-io/milvus#25422] [Q&A and General discussion]
can we enable the autoscaling on milvus cluster(without manually setting up the replicas) which is deployed on GCP

**Comment by yhmo** [comment]
According to the doc, currently, Milvus only supports scaling in and out manually.
https://milvus.io/docs/scaleout.md#Scale-a-Milvus-cluster

**Reply by divyajaincs** [reply]
@yhmo yeah but there it is also mentioned the auto scaling support based on resources so it's already in the latest release or it's coming in upcoming release

**Comment by xiaofan-luan** [comment]
> @yhmo yeah but there it is also mentioned the auto scaling support based on resources so it's already in the latest release or it's coming in upcoming release

Currently we don't have plan to support auto scale but it shouldn't be too hard to support by KNative

**Reply by divyajaincs** [reply]
@xiaofan-luan Any Documentation on how we can autoscale milvus with Knative?

**Comment by DimasAdinugroho** [comment]
Can we use HPA to support auto scaling for Milvus? @xiaofan-luan

**Reply by xiaofan-luan** [reply]
I think this a topic to be discussed.
If simply consider about memory i think HPA is doable.
@LoveEachDay any ideas?

**Reply by LoveEachDay** [reply]
@divyajaincs You can use hpa to scaling out/down based on querynode memory usage.

**Reply by DimasAdinugroho** [reply]
@LoveEachDay I tried by setting the `scaleTargetRef` kind to `Deployment`, but it seems that the `Deployment` workload is controlled by `MilvusOperator`, so the replica is automatically rollback to the current value or is there another way?

**Comment by chikymanzanares** [comment]
During peak search times, we experience an issue where a high volume of requests causes the search to go down. Do you have any suggestions or temporary solutions we can implement until official support is provided?

**Reply by xiaofan-luan** [reply]
> During peak search times, we experience an issue where a high volume of requests causes the search to go down. Do you have any suggestions or temporary solutions we can implement until official support is provided?

Scale up or scale out? what is your current QPS and resource usage?
Move to zilliz cloud might be another option

**Reply by chikymanzanares** [reply]
search latency aprox 7.8k
![Screenshot 2024-06-10 at 14 52 59](https://github.com/milvus-io/milvus/assets/32399434/310ba97f-9af9-4f0d-a7ee-094aca5a4424)

**Reply by chikymanzanares** [reply]
scale up, when we have many request, we have to increase query nodes, now we have 16, we are stable, but for peaks we need to scale automatically

**Reply by navneetrajput06** [reply]
Hi team,
Do we have autoscaling available in Milvus 2.4.4 ?

**Reply by xiaofan-luan** [reply]
cc @zwd1208

**Reply by codingjaguar** [reply]
@navneetrajput06 I checked recently Milvus auto scale is not available the best approach is still HPA based on query node metrics like memory usage. And Zilliz cloud (fully managed Milvus) is a good alternative to consider.

**Comment by aswathinj-pai** [comment]
is it possible to enable HPA in milvus standalone?

**Reply by codingjaguar** [reply]
> is it possible to enable HPA in milvus standalone?

HPA is a concept in K8s i don't think it would work for Milvus Standalone on docker.
